### PR TITLE
WIP: Update go-amino (wire) to develop

### PR DESCRIPTION
**Pull request by melekes**
Refs #1186

### CODE REVIEW

**Comment by ebuchman** [comment]
GL

File: benchmarks/codec_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func BenchmarkEncodeStatusWire(b *testing.B) {
 
 	counter := 0
 	for i := 0; i < b.N; i++ {
-		jsonBytes := wire.JSONBytes(status)
+		jsonBytes := amino.JSONBytes(status)
```
Comment (odeke-em) [inline benchmarks/codec_test.go]: Does this code compile?

**Comment by odeke-em** [comment]
Nice initiative, taking one for the team @melekes 🍻

File: types/priv_validator.go [hunk 2]
```diff
@@ -55,30 +55,30 @@ type PrivValidator2 interface {
 
 type TestSigner interface {
 	Address() cmn.HexBytes
-	PubKey() crypto.PubKey
```
Comment (melekes) [inline types/priv_validator.go]: should we use concrete types here?

File: types/priv_validator.go [hunk 3]
```diff
@@ -55,30 +55,30 @@ type PrivValidator2 interface {
...
+	PrivKey crypto.PrivKeyEd25519
 }
 
 func (ds *DefaultTestSigner) Address() cmn.HexBytes {
 	return ds.PubKey().Address()
 }
 
-func (ds *DefaultTestSigner) PubKey() crypto.PubKey {
-	return ds.PrivKey.PubKey()
+func (ds *DefaultTestSigner) PubKey() crypto.PubKeyEd25519 {
```
Comment (melekes) [inline types/priv_validator.go]: same question

File: types/priv_validator.go [hunk 4]
```diff
@@ -99,16 +99,16 @@ type PrivValidator interface {
...
 type PrivValidatorFS struct {
-	Address       Address          `json:"address"`
-	PubKey        crypto.PubKey    `json:"pub_key"`
-	LastHeight    int64            `json:"last_height"`
-	LastRound     int              `json:"last_round"`
-	LastStep      int8             `json:"last_step"`
-	LastSignature crypto.Signature `json:"last_signature,omitempty"` // so we dont lose signatures
-	LastSignBytes cmn.HexBytes     `json:"last_signbytes,omitempty"` // so we dont lose signatures
+	Address       Address                 `json:"address"`
+	PubKey        crypto.PubKey           `json:"pub_key"`
```
Comment (melekes) [inline types/priv_validator.go]: should be `PubKeyEd25519` probably ...

File: types/priv_validator.go [hunk 5]
```diff
@@ -282,7 +282,7 @@ func (privVal *PrivValidatorFS) checkHRS(height int64, round int, step int8) (bo
 				return false, errors.New("Step regression")
 			} else if privVal.LastStep == step {
 				if privVal.LastSignBytes != nil {
-					if privVal.LastSignature == nil {
+					if len(privVal.LastSignature) == 0 {
```
Comment (melekes) [inline types/priv_validator.go]: `IsZero`

File: types/priv_validator/json.go [hunk 6]
```diff
@@ -17,26 +17,15 @@ type PrivValidator = types.PrivValidator2
...
 
 // PrivKey implements Signer
-type PrivKey crypto.PrivKey
+type PrivKey crypto.PrivKeyEd25519
 
 // Sign - Implements Signer
-func (pk PrivKey) Sign(msg []byte) (crypto.Signature, error) {
-	return crypto.PrivKey(pk).Sign(msg), nil
+func (pk PrivKey) Sign(msg []byte) (crypto.SignatureEd25519, error) {
+	return crypto.PrivKeyEd25519(pk).Sign(msg).(crypto.SignatureEd25519), nil
```
Comment (melekes) [inline types/priv_validator/json.go]: don't like this ... looks bad

**Comment by melekes** [comment]
https://github.com/tendermint/tendermint/pull/1347

**Pull request closed without merging**